src/main/kotlin/com/example/auctiondemo/aggregate/ProductAggregate.kt

```diff
@@ -0,0 +1,105 @@
+package com.example.auctiondemo.aggregate
+
+import com.example.auctiondemo.api.command.*
+import com.example.auctiondemo.domain.BidStatus
+import com.example.auctiondemo.repository.AuctionProductRepository
+import org.axonframework.commandhandling.CommandHandler
+import org.axonframework.eventsourcing.EventSourcingHandler
+import org.axonframework.modelling.command.AggregateIdentifier
+import org.axonframework.modelling.command.AggregateLifecycle
+import org.axonframework.spring.stereotype.Aggregate
+import org.springframework.beans.factory.annotation.Autowired
+import java.math.BigDecimal
+import java.util.Date
+
+@Aggregate
+class ProductAggregate() {
+
+    @AggregateIdentifier
+    private lateinit var productId: String
+    private lateinit var name: String
+    private lateinit var description: String
+    private lateinit var startPrice: BigDecimal
+    private lateinit var endedDateTime: Date
+//    private var endedDateTime: Date
+    private lateinit var status: BidStatus
+    private lateinit var currentBidOwner: String
+    private lateinit var currentHighestBid: BigDecimal
+
+    @Autowired
+    private lateinit var productRepository: AuctionProductRepository
+
+    @CommandHandler
+    constructor(command: CreateProductCommand) : this(){
+        AggregateLifecycle.apply(
+            CreateProductEvent(
+            command.productId,
+            command.name,
+            command.description,
+            command.startPrice,
+            BidStatus.NONE)
+        )
+    }
+
+    @EventSourcingHandler
+    fun on (event: CreateProductEvent){
+        productId = event.productId
+        name = event.name
+        description = event.description
+        startPrice = event.startPrice
+    }
+
+    @CommandHandler
+    fun handle(command : StartAuctionCommand) : String{
+        val date = productRepository.findById(command.productId).block()!!.endedDateTime
+        if(date == null){
+            val endedDate = Date(System.currentTimeMillis() + (command.durationMin * 60 * 1000))
+            println(endedDate)
+            AggregateLifecycle.apply(StartAuctionEvent(command.productId, endedDate, BidStatus.STARTED))
+            return "Auction start!!"
+        } else {
+            return "The auction is already started. I can't be auctioned twice nor thrice."
```

Comment:
ควรจะใช้การ throw exception แทนการส่ง message เช่น
`throw new IllegalStateException("The auction was already started.")`